### PR TITLE
Set starting size for changed text builder using old text

### PR DESCRIPTION
**Pull request by arkalyanms**
For perf epic [1611109](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1611109)

### CODE REVIEW

File: src/Compilers/Core/Portable/Text/ChangedText.cs [hunk 1]
```diff
@@ -278,7 +278,7 @@ protected override TextLineCollection GetLinesCore()
             }
 
             // compute line starts given changes and line starts already computed from previous text
-            var lineStarts = ArrayBuilder<int>.GetInstance();
+            var lineStarts = ArrayBuilder<int>.GetInstance(oldText.Lines.Count);
```
Comment (333fred) [inline src/Compilers/Core/Portable/Text/ChangedText.cs:281]:
```suggestion
            var lineStarts = ArrayBuilder<int>.GetInstance(oldLineInfo.Count);
```

**Review by jasonmalinowski — APPROVED** [review]
Seems reasonable; I don't know if there's a deeper problem for that bug of why we're seeing so many ChangedTexts in the first place, but I don't see this change causing any harm (or not likely, at least.)

**Comment by 333fred** [comment]
@arkalyanms changes in the compilers layer require 2 compiler sign offs.

**Review by 333fred — APPROVED** [review]
LGTM (commit 2). @dotnet/roslyn-compiler for a second review.

**Comment by arkalyanms** [comment]
> Seems reasonable; I don't know if there's a deeper problem for that bug of why we're seeing so many ChangedTexts in the first place, but I don't see this change causing any harm (or not likely, at least.)

Yes I am planning to keep the actual bug open and alleviate just the perfepic focus.

**Review by chsienki — APPROVED** [review]

**Review by RikkiGibson — APPROVED** [review]

**Comment by arkalyanms** [comment]
@genlu fyi on another 17.4 fix for a perf epic which we will take for approvals when Preview 4 opens up.

**Comment by arkalyanms** [comment]
Its in main now.

**Pull request merged**